pkg/apis/workspaces/v1alpha2/devworkspaceTemplateSpec.go

```diff
@@ -12,6 +14,15 @@ type DevWorkspaceTemplateSpec struct {
 
 // +devfile:overrides:generate
 type DevWorkspaceTemplateSpecContent struct {
+	// Map of implementation-dependant free-form YAML attributes.
+	// Attribute values can be referenced throughout the devfile in string type fields in the form {{attribute-key}}
+	// except for schemaVersion and metadata. Exception to the string field include element's key identifiers(command id,
+	// component name, endpoint name, project name, etc.) and string enums(command group kind, endpoint exposure, etc.)
+	// +optional
+	// +patchStrategy=merge
+	// +devfile:overrides:include:omitInPlugin=true,description=Overrides of attributes encapsulated in a parent devfile.
+	Attributes attributes.Attributes `json:"attributes,omitempty" patchStrategy:"merge"`
```

Comment:
@amisevsk metadata attributes can store information about data that is not required to be used and substituted in other devfile fields. FWIW, I had the same question as you regd removal of the metadata attribute.

Reply:
With that in mind then, are the global attributes intended to be _only_ used for substitution? I was kind of imagining using the global attributes for both, at which point `.metadata.attributes` is redundant.

Reply:
Unless we're explicitly forbidding the global attributes introduced here from storing non-replacement attributes, then I think it doesn't make sense to have both -- it'll just be a source of confusion. If global attributes to be used *just* for replacement, then why isn't their type `map[string]string`?

Reply:
@amisevsk I've talked with Elson. We're definitely interested in keeping metadata attributes as they will be used for information pertaining to devfiles. And they've been used in the past by both odo and Console devfiles.

With that being said, as I see Global Attributes at this point to be used for replacement, I am changing the type to `map[string]string` as you suggested to avoid confusion.

Reply:
If the intended usage of the global attributes is just for replacement, what about just calling them such? E.g.:

```yaml
schemaVersion: 2.1.0
metadata:
  name: my-devfile
  attributes:
    # free-form yaml
substitutions:
  # string: string
```

Reply:
@metlos  Not sure if others will agree on `substitutions` but open to suggestions 😄 
@amisevsk any opinion regd the rename?

Reply:
Addressed type in new commit :)